### PR TITLE
[CCI-1818] Add documentation for CCM EBS shared cost allocation

### PR DESCRIPTION
**Pull request by kaankoseler**
### What does this PR do? What is the motivation?
We are releasing a new feature in Cloud Cost Management: shared cost allocation for non-local EBS volumes used as Persistent Volumes in Kubernetes clusters.

### Merge instructions
Ideally we will release these in quick succession, with the product feature releasing to all customers first. It's currently org 2 only. We would like to have control of the merge.

- [ ] Please merge after reviewing

### Additional notes

### CODE REVIEW

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 1]
```diff
@@ -30,16 +32,27 @@ Datadog Cloud Cost Management (CCM) automatically allocates EC2 compute cost in
...
 
 Next, Datadog looks at all of the pods running on that node for the day. The cost of the node is allocated to the pod based on the resources it has used and the length of time it ran. This calculated cost is enriched with all of the pod's tags.
 
 **Note**: Only _tags_ from pods and nodes are added to cost metrics. To include labels, enable labels as tags for [nodes][7] and [pods][8].
 
+#### Persistent volume storage
+
+For Kubernetes Persistent Volume storage allocation, Persistent Volumes (PV), Persistent Volume Claims (PVC), nodes, and pods are joined with their associated EBS volume costs. All associated PV, PVC, node, and pod tags are added to the EBS volume cost line items.
+
+Next, Datadog looks at all of the pods that claimed the volume on that day. The cost of the volume is allocated to the pod based on the resources it has used and the length of time it ran. These resources
```
Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:52]:
```suggestion
Next, Datadog looks at all of the pods that claimed the volume on that day. The cost of the volume is allocated to a pod based on the resources it used and the length of time it ran. These resources
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 2]
```diff
@@ -30,16 +32,27 @@ Datadog Cloud Cost Management (CCM) automatically allocates EC2 compute cost in
...
+include the provisioned capacity for Storage, IOPS, and Throughput. This allocated cost is enriched with all of the pod's tags.
```
Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:53]:
```suggestion
include the provisioned capacity for storage, IOPS, and throughput. This allocated cost is enriched with all of the pod's tags.
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 3]
```diff
@@ -52,24 +65,33 @@ ECS tasks that run on Fargate are already fully allocated in the CUR. CCM enrich
 
 ### Everything else
 
-Any cost other than EC2, computed for instances hosting Kubernetes pods or ECS tasks, is given the same value and tags as the source metric, `aws.cost.amortized`.
+Cost other than EC2/EBS, computed for instances hosting Kubernetes pods or ECS tasks, is given the same value and tags as the source metric, `aws.cost.amortized`.
```
Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:68]:
```suggestion
All other costs (costs that are not related to EC2 or EBS, and are not computed for instances hosting Kubernetes pods or ECS tasks) are given the same value and tags as the source metric, `aws.cost.amortized`.
```

@kaankoseler I'm trying to reword this sentence to make it more clear. Can you please check that my change doesn't change the intended meaning of the sentence?

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 4]
```diff
@@ -52,24 +65,33 @@ ECS tasks that run on Fargate are already fully allocated in the CUR. CCM enrich
...
+
+### Compute
 
 - Usage: Cost of memory and cpu being used or reserved by workloads.
 - Workload idle: Cost of memory and cpu that is being reserved and allocated but not used by workloads.
 - Cluster idle: Cost of memory and cpu not reserved by workloads in a cluster.
 
+### Persistent volumes
+
+- Usage: Cost of provisioned IOPS, Throughput, or Storage being used by workloads. For Storage, this is based on the maximum amount of volume storage used that day. For IOPS and Throughput, this is based on the average amount used that day.
```
Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:83]:
```suggestion
- Usage: Cost of provisioned IOPS, throughput, or storage being used by workloads. Storage cost is based on the maximum amount of volume storage used that day. IOPS and throughput costs are based on the average amount used that day.
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 5]
```diff
@@ -52,24 +65,33 @@ ECS tasks that run on Fargate are already fully allocated in the CUR. CCM enrich
...
+- Workload idle: Cost of provisioned IOPS, Throughput, or Storage not being used by workloads. For Storage, this is based on the maximum amount of storage used that day. For IOPS and Throughput, this is based on the average amount used that day.
```
Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:84]:
```suggestion
- Workload idle: Cost of provisioned IOPS, throughput, or storage not being used by workloads. Storage cost is based on the maximum amount of storage used that day. IOPS and throughput costs are based on the average amount used that day.
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 6]
```diff
@@ -52,24 +65,33 @@ ECS tasks that run on Fargate are already fully allocated in the CUR. CCM enrich
...
+- Cluster idle: Cost of provisioned IOPS, Throughput, or Storage for volumes not claimed by any pods that day.
```
Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:85]:
```suggestion
- Cluster idle: Cost of provisioned IOPS, throughput, or storage for volumes not claimed by any pods that day.
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 7]
```diff
@@ -96,6 +118,7 @@ Datadog consolidates and applies additional tags from various sources to cost me
 | `allocated_resource:cpu`                   | Cost of CPU resources. *Only available for `.shared.resources.allocated` metrics.*                                            |
 | `allocated_resource:memory`                | Cost of Memory resources. *Only available for `.shared.resources.allocated` metrics.*                                         |
 | `allocated_resource:managed_service_fee`   | Cost of cloud provider managed service fees. *Only available for `.shared.resources.allocated` metrics.*           |
+| `allocated_resource:persistent_volume`     | Cost of non-local EBS Volumes used as Persistent Volumes in Kubernetes. *Only available for `.shared.resources.allocated` metrics.*           |
```
Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:121]:
```suggestion
| `allocated_resource:persistent_volume`     | Cost of non-local EBS volumes used as Persistent Volumes in Kubernetes. *Only available for `.shared.resources.allocated` metrics.*           |
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 8]
```diff
@@ -119,6 +142,17 @@ In addition to ECS task tags, the following out-of-the-box tags are applied to c
...
+
+| Out-of-the-box tag                      | Description                                                                                                                                 |
+| ---                                     |---------------------------------------------------------------------------------------------------------------------------------------------|
+| `persistent_volume_reclaim_policy`      | The Kubernetes Reclaim Policy on the Persistent Volume.                                                                                     |
+| `storage_class_name`                    | The Kubernetes Storage Class used to instantiate the Persistent Volume.                                                                     |
+| `volume_mode`                           | The Volume Mode of the Persistent Volume.                                                                                                   |
+| `ebs_volume_type`                       | The type of the AWS EBS volume. *Can be `gp3`, `gp2`, and more.*                                                                            |
```
Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:154]:
```suggestion
| `ebs_volume_type`                       | The type of the AWS EBS volume. Can be `gp3`, `gp2`, or others.                                                                            |
```

**Review by cswatt — CHANGES_REQUESTED** [review]
Mostly small edits, though I'd like verification on one edit to make sure I didn't change the intended meaning.

File: content/en/cloud_cost_management/container_cost_allocation.md (hunk 3 above)
Comment (kaankoseler) [reply content/en/cloud_cost_management/container_cost_allocation.md:68]: Hi @cswatt thank you for the review.

This is a tough sentence to work with, I think. The revision does seem to lose the meaning somewhat.  
The idea that we are trying to convey is that this subset of costs is what's being targeted with the feature:
`EC2/EBS costs for EC2 instances hosting Kubernetes pods or ECS tasks`


Any cost that does not fall under that umbrella is considered `Everything else` and left as is with the source metric. 
So we somehow need to keep that sentence above as one coherent whole, IMO.

Comment (enemyghost) [reply content/en/cloud_cost_management/container_cost_allocation.md:68]: As the list of allocated cost types is growing, this sentence might just not be scalable 😆 I wonder if we even need to re-list all the costs above. One option is just drop the explanation:
"All other costs are given the same value and tags as the source metric, `aws.cost.amortized`."

Or, if we want an explanation, to split it up into multiple sentences:
"All other costs are given the same value and tags as the source metric, `aws.cost.amortized`. This includes costs for all other AWS services, as well as EC2 and EBS costs not associated with container orchestration."

Comment (kaankoseler) [reply content/en/cloud_cost_management/container_cost_allocation.md:68]: I would prefer the former, personally. It should be pretty obvious from the sections and the preceding text of the document

Comment (kaankoseler) [reply content/en/cloud_cost_management/container_cost_allocation.md:68]: @cswatt what do you think about just keeping this as:
`"All other costs are given the same value and tags as the source metric, aws.cost.amortized."`

Comment (cswatt) [reply content/en/cloud_cost_management/container_cost_allocation.md:68]: How about "The information on this page pertains to EC2/EBS costs for EC2 instances hosting Kubernetes pods or ECS tasks. All other costs are given the same value and tags as the source metric, `aws.cost.amortized`." —does that work?

Comment (kaankoseler) [reply content/en/cloud_cost_management/container_cost_allocation.md:68]: Yes, that works! cc @enemyghost

Comment (kaankoseler) [reply content/en/cloud_cost_management/container_cost_allocation.md:68]: I talked to @enemyghost offline, can we slightly modify to say `... pertains to EC2 Compute and EBS costs...`?

Comment (cswatt) [inline content/en/cloud_cost_management/container_cost_allocation.md:68]:
```suggestion
The information on this page pertains to EC2 compute and EBS costs for EC2 instances hosting Kubernetes pods or ECS tasks. All other costs are given the same value and tags as the source metric, `aws.cost.amortized`.
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**